cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/services/AppFabricServerTest.java

```diff
@@ -105,4 +97,24 @@ public EndpointStrategy get() {
     socket.startHandshake();
     appFabricServer.stopAndWait();
   }
+
+  private Injector getInjector(CConfiguration cConf, @Nullable SConfiguration sConf) {
+    Injector injector = Guice.createInjector(new AppFabricTestModule(cConf, sConf));
+    injector.getInstance(TransactionManager.class).startAndWait();
+    // Register the tables before services will need to use them
+    StructuredTableAdmin tableAdmin = injector.getInstance(StructuredTableAdmin.class);
+    StructuredTableRegistry structuredTableRegistry = injector.getInstance(StructuredTableRegistry.class);
+    try {
+      structuredTableRegistry.initialize();
+    } catch (IOException e) {
+      throw new RuntimeException(e);
+    }
+    try {
+      StoreDefinition.createAllTables(tableAdmin, structuredTableRegistry);
+    } catch (IOException | TableAlreadyExistsException e) {
+      throw new RuntimeException("Failed to create the system tables", e);
+    }
+    injector.getInstance(DatasetService.class).startAndWait();
```

Comment:
DatasetService and TransactionManager have to be stopped, right?

Reply:
Added the stop method